tests/unittests/test_tabular_bagstack.py

```diff
@@ -0,0 +1,176 @@
+    # If True, run a faster benchmark (subsample training sets, less epochs, etc),
+    # otherwise we run full benchmark with default AutoGluon settings.
+    # performance_value warnings are disabled when fast_benchmark = True.
+
+    #### If fast_benchmark = True, can control model training time here. Only used if fast_benchmark=True ####
+    if fast_benchmark:
+        subsample_size = 100
+        nn_options = {'num_epochs': 3} 
+        gbm_options = {'num_boost_round': 30}
+        hyperparameters = {'GBM': gbm_options, 'NN': nn_options}
+        time_limits = 60
+
+    fit_args = {
+        'num_bagging_folds': num_bagging_folds,
+        'stack_ensemble_levels': stack_ensemble_levels,
+        'hyperparameter_tune': hyperparameter_tune,
+        'verbosity': verbosity,
+    }
+    if hyperparameters is not None:
+        fit_args['hyperparameters'] = hyperparameters
+    if time_limits is not None:
+        fit_args['time_limits'] = time_limits
+    ###################################################################
+    run_tabular_benchmarks(fast_benchmark=fast_benchmark, subsample_size=subsample_size, perf_threshold=perf_threshold,
+                           seed_val=seed_val, fit_args=fit_args)
+
+
+def test_tabular_stack1():
+    ############ Benchmark options you can set: ########################
+    stack_ensemble_levels = 1
+    num_bagging_folds = 0
+    perf_threshold = 1.1 # How much worse can performance on each dataset be vs previous performance without warning
+    seed_val = 0 # random seed
+    subsample_size = None
+    hyperparameter_tune = False
+    verbosity = 4 # how much output to print
+    hyperparameters = None
+    time_limits = None
+    fast_benchmark = True # False
+    # If True, run a faster benchmark (subsample training sets, less epochs, etc),
+    # otherwise we run full benchmark with default AutoGluon settings.
+    # performance_value warnings are disabled when fast_benchmark = True.
+
+    #### If fast_benchmark = True, can control model training time here. Only used if fast_benchmark=True ####
+    if fast_benchmark:
+        subsample_size = 100
+        nn_options = {'num_epochs': 3} 
+        gbm_options = {'num_boost_round': 30}
+        hyperparameters = {'GBM': gbm_options, 'NN': nn_options}
+        time_limits = 60
+
+    fit_args = {
+        'num_bagging_folds': num_bagging_folds,
+        'stack_ensemble_levels': stack_ensemble_levels,
+        'hyperparameter_tune': hyperparameter_tune,
+        'verbosity': verbosity,
+    }
+    if hyperparameters is not None:
+        fit_args['hyperparameters'] = hyperparameters
+    if time_limits is not None:
+        fit_args['time_limits'] = time_limits
+    ###################################################################
+    run_tabular_benchmarks(fast_benchmark=fast_benchmark, subsample_size=subsample_size, perf_threshold=perf_threshold,
+                           seed_val = seed_val, fit_args=fit_args)
+
+
+def test_tabular_stack2():
```

Comment:
This is thorough but perhaps we could just check if the stack_ensemble_levels is set to 0 because of num_bagging_folds=0 instead of actually running it. We could do this by splitting .fit internal logic into two stages: Handling params and the call to learner.fit(), we could avoid running the second stage and just check if stack_ensemble_levels was set to 0.

Reply:
you don't think we'd ever support stacking without bagging in the future? If so I can just remove these tests {test_tabular_stack1() and test_tabular_stack2()} from being run at the moment.